swan-lake/learn-the-language/language-basics.md

```diff
@@ -473,6 +473,98 @@ function nameToString(Name nm) returns string {
  
 In this way, the ``is`` operator in a condition causes the declared type to be narrowed.
 
+## RegExp Type
+
+Ballerina provides first-class support for regular expressions, which are represented by the ``RegExp`` built-in type defined in the ``lang.regexp`` module.
+
+You can declare regular expressions in two ways.
+
+Using regular expression template literal:
+```ballerina
+string:RegExp reg = re `abc+`;
+```
+In this approach, you can use the ``re`` keyword to define a regular expression pattern using a template literal. This allows for easy expression of regular expressions in a way that is similar to other languages.
+
+Using the fromString method in the ``lang.regexp`` module.
+```ballerina
+import lang.regexp;
+string quantiferStr = "{2,3}";
+string stringPattern = string `abc${quantiferStr}`;
+regexp:RegExp reg = check regexp:fromString(stringPattern);
+```
```

Comment:
Shall we add new lines as appropriate?

````suggestion
Using the fromString method in the ``lang.regexp`` module.

```ballerina
import lang.regexp;

string quantiferStr = "{2,3}";
string stringPattern = string `abc${quantiferStr}`;
regexp:RegExp reg = check regexp:fromString(stringPattern);
```

````